devmand/gateway/src/devmand/channels/cli/SshSocketReader.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2019-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+
+#include <devmand/channels/cli/SshSocketReader.h>
+#include <devmand/channels/cli/SshSessionAsync.h>
+#include <event2/event.h>
+#include <thread>
+#include <chrono>
+
+using devmand::channels::cli::sshsession::SshSessionAsync;
+
+void sshReadNotificationThread(struct event_base *base);
+void sshReadNotificationThread(struct event_base *base) {
+    while (true) {
+        int rv = event_base_dispatch(base);
+        if (rv == 1) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
What is this for?

Reply:
If event_base_dispatch has no events to listen to (there would be no CLI devices) it will end. Not sure if this scenario is possible. Just to be sure, we check the condition that there are no events registered (rv == 1) and we wait 1 second to retry to start the event_base_dispatch loop (if any device has registered in the meantime it will be picked-up by the loop). The 1 second is a back-off as we don't need to restart the libevent loop right away.